cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -386,17 +391,165 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
+
+        Map<Integer, Map<String, Object>> tempMap = new HashMap<>();
+
+        for (Object rawBrokerLoad : brokerLoadArray) {
+            JsonObject brokerLoad = (JsonObject) rawBrokerLoad;
+
+            Map<String, Object> brokerLoadMap = new HashMap<>();
+
+            for (CruiseControlLoadParameters intParam : CruiseControlLoadParameters.getIntegerParameters()) {
+                if (brokerLoad.containsKey(intParam.getCruiseControlKey())) {
+                    brokerLoadMap.put(intParam.getStrimziKey(), brokerLoad.getInteger(intParam.getCruiseControlKey()));
+                }
+            }
+
+            for (CruiseControlLoadParameters doubleParam : CruiseControlLoadParameters.getDoubleParameters()) {
+                if (brokerLoad.containsKey(doubleParam.getCruiseControlKey())) {
+                    brokerLoadMap.put(doubleParam.getStrimziKey(), brokerLoad.getDouble(doubleParam.getCruiseControlKey()));
+                }
+            }
+
+            int brokerID = brokerLoad.getInteger(CruiseControlRebalanceKeys.BROKER_ID.getKey());
+            tempMap.put(brokerID, brokerLoadMap);
+
+        }
+
+        return tempMap;
+
+    }
+
+    /**
+     * Converts the supplied before and after broker load arrays into a map linking from broker ID integer to a map linking
+     * from load parameter to an array of [before, after, difference]. The load paramters included in the map are dictated
+     * by the values in he {@link CruiseControlLoadParameters} enum.
+     *
+     * @param brokerLoadBeforeArray The JSONArray of broker load JSONObjects, for before the optimization proposal is applied,
+     *                              returned by the Cruise Control rebalance endpoint.
+     * @param brokerLoadAfterArray The JSONArray of broker load JSONObjects, for after the optimization proposal is applied,
+     *                             returned by the Cruise Control rebalance endpoint.
+     * @return A map linking from broker ID integer to a map of load parameter to [before, after, difference] arrays.
+     */
+    protected static Map<Integer, Map<String, Object>> createBeforeAfterLoadMap(JsonArray brokerLoadBeforeArray, JsonArray brokerLoadAfterArray) {
+
+        // There is no guarantee that the brokers are in the same order in both the before and after arrays.
+        // Therefore we need to convert them into maps indexed by broker ID so we can align them later for the comparison.
+        Map<Integer, Map<String, Object>> loadBeforeMap = extractLoadParameters(brokerLoadBeforeArray);
+        Map<Integer, Map<String, Object>> loadAfterMap = extractLoadParameters(brokerLoadAfterArray);
+
+        if (loadBeforeMap.size() != loadAfterMap.size()) {
+            throw new RuntimeException("Broker data was missing from the load before/after information");
+        }
+
+        Map<Integer, Map<String, Object>> brokersBeforeAfterStats = new HashMap<>();
```

Comment:
```suggestion
        Map<Integer, Map<String, Object>> brokersStats = new HashMap<>();
```
I think it can be assumed that the brokerStats are in the context of before/after info